functests/ptp/ptp.go

```diff
@@ -108,4 +96,437 @@ var _ = Describe("ptp", func() {
+			})
+		})
+		AfterEach(func() {
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(configPtpAPIPath + "/test").Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP slave config recovery Error"))
+			waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: slave")
+		})
+	})
 })
+
+func podRole(runningPod v1core.Pod, role string) bool {
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{
+		LabelSelector: role,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	for NodeNumber := range nodeList.Items {
+		if runningPod.Spec.NodeName == nodeList.Items[NodeNumber].Name {
+			return true
+		}
+	}
+	return false
+}
+
+func execCommand(pod v1core.Pod, command []string) (string, bytes.Buffer) {
+	var buf bytes.Buffer
+	req := client.Client.CoreV1Interface.RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&v1core.PodExecOptions{
+			Container: pod.Spec.Containers[0].Name,
+			Command:   command,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+
+	Eventually(func() error {
+		exec, err := remotecommand.NewSPDYExecutor(client.Client.Config, "POST", req.URL())
+		Expect(err).ToNot(HaveOccurred())
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: &buf,
+			Stderr: os.Stderr,
+			Tty:    true,
+		})
+		if err == nil {
+			commandOut := buf.String()
+			if len(commandOut) > 0 {
+				return nil
+			}
+			return fmt.Errorf("No content in command output")
+		}
+		return err
+	}, 1*time.Minute, 1*time.Second).Should(BeNil())
+
+	commandOut := buf.String()
+	return commandOut, buf
+}
+
+func compareOffsetTime(timeDiff string) bool {
+	var timeStampList []int
+	for _, line := range strings.Split(timeDiff, "\n") {
+		if strings.Contains(line, "openshift_ptp_max_offset_from_master") && !strings.Contains(line, "# ") {
+			splitedLine := strings.Split(line, " ")
+			offcetFromMaster, err := strconv.Atoi(strings.Trim(splitedLine[len(splitedLine)-1], "\r"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(offcetFromMaster).To(BeNumerically("<", 100))
+			Expect(offcetFromMaster).To(BeNumerically(">", -100))
+			timeStampList = append(timeStampList, offcetFromMaster)
+		}
+	}
+	Expect(len(timeStampList)).To(BeNumerically("==", 2))
+	return true
+}
+
+func getPodLogs(pod v1core.Pod) (string, error) {
+	podLogOpts := v1core.PodLogOptions{}
+	req := client.Client.CoreV1Interface.Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream()
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+
+	return str, nil
+}
+
+func netParseEthtoolTimeStampFeature(cmdOut *bytes.Buffer) bool {
```

Comment:
EthTool. Also, what is the purpose of this function?

Reply:
Added comment to function and renamed as you suggested